### PR TITLE
Add healthcheck for default localhost 8008 port on /health endpoint.

### PR DESCRIPTION
**Pull request by maquis196**
* Updated docker readme for healthcheck related options

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -162,3 +162,28 @@ docker build -t matrixdotorg/synapse -f docker/Dockerfile .
 
 You can choose to build a different docker image by changing the value of the `-f` flag to
 point to another Dockerfile.
+
+## Disabling the healthcheck
+
+If you are using a non-standard port or tls inside docker you can disable the healthcheck
+whilst running the above docker run commands.
```
Comment (babolivier) [inline docker/README.md:169]:
```suggestion
whilst running the above `docker run` commands. 
```

File: changelog.d/8147.docker [hunk 2]
```diff
@@ -0,0 +1 @@
+Added curl for healthcheck support and readme updates for the change. Added by @maquis196.
```
Comment (babolivier) [inline changelog.d/8147.docker:1]:
```suggestion
Added curl for healthcheck support and readme updates for the change. Contributed by @maquis196.
```

**Review by babolivier — APPROVED** [review]
Thanks for your contribution! This looks good to me, let's merge it :)

**Comment by maquis196** [comment]
updated the docker-compose test to make it clear you'll need version 2.1 at least, plus a rebase to see if that fixes the failing CI.

File: docker/README.md [hunk 3]
```diff
@@ -162,3 +162,31 @@ docker build -t matrixdotorg/synapse -f docker/Dockerfile .
...
+
+If you wish to point the healthcheck at a different port with docker command, add the following
+
+```
+  --health-cmd 'curl -fSs http://localhost:1234/health'
+```
+
+## Setting the healthcheck in docker-compose file
+
+You can add the following to set a custom healthcheck in a docker compose file. You'll need version >2.1 for this to work.
```
Comment (babolivier) [inline docker/README.md:184]:
```suggestion
You can add the following to set a custom healthcheck in a docker compose file.
You'll need version >2.1 for this to work. 
```

**Comment by schildbach** [comment]
Just for reference, since 0.20.x I'm using this as my health check from within docker-compose:

```
    healthcheck:
      test: ["CMD-SHELL", "curl --fail --silent http://localhost:8008/_synapse/admin/v1/server_version || exit 1"]
```

**Comment by maquis196** [comment]
> Just for reference, since 1.20.x I'm using this as my health check from within docker-compose:
> 
> ```
>     healthcheck:
>       test: ["CMD-SHELL", "curl --fail --silent http://localhost:8008/_synapse/admin/v1/server_version || exit 1"]
> ```

out of pure interest, why server_version and not health?

**Comment by schildbach** [comment]
Ah, I didn't know this endpoint exists. Was using server_version for about a year or so, and it worked.

**Comment by SuperSandro2000** [comment]
This healthcheck should be reverted. It just delays the image startup by approximately one minute of which the container most likely waits 50 seconds for no reason.

**Comment by maquis196** [comment]
> This healthcheck should be reverted. It just delays the image startup by approximately one minute of which the container most likely waits 50 seconds for no reason.

synapse needs to be running for the healthcheck to work, so what would be waiting a minute? seeing "UNHEALTHY" in a `docker ps` doesn't mean synapse is uncontactable right? 

if i start synapse, i can run the above curl manually within seconds suggesting that synapse is up and running, but docker will wait a minute to say its healthcheck has passed

**Comment by SuperSandro2000** [comment]
> synapse needs to be running for the healthcheck to work, so what would be waiting a minute?

Docker tries the healthcheck once and if synapse is not up yet it waits a minute and then tries again which is annoying.

> seeing "UNHEALTHY" in a docker ps doesn't mean synapse is uncontactable right?

If you are using a reverse proxy like traefik it gets removed from the connection pool because it is unhealthy.

**Comment by maquis196** [comment]
> If you are using a reverse proxy like traefik it does not get added to the connection pool because it is unhealthy.

Interesting, so the solution here would be to add '--start-period=5' and maybe change interval to 15, since it tries immediately which is about a couple of seconds too quick

**Comment by SuperSandro2000** [comment]
That still introduces some seconds delay where you just wait until its done. In my opinion the better solution would be to just remove it as it serves little to no purpose in a single container setup.

**Comment by maquis196** [comment]
If the powers that be want to remove it, then great, as long as they leave curl in, bottom line is though you can easily disable it on your own setup if you don't feel the need to have it, instructions are in the readme, ill update the new PR with docker-compose instructions on disabling too since i forgot those.

**Pull request merged**